### PR TITLE
Added tags for tools, armor and weapons

### PR DESCRIPTION
**Pull request by issork**
I've been looking for a tag for hoes and came across #6152.

Only differences to what was attempted in the above pull requests are;
- no ranged tag for bows/crossbows/tridents
- bows/crossbows are separated and not both categorized into bows
- shield also added to armor tag

Please let me know if there are any further desired tags or if any of the new tags are obsolete.

### CODE REVIEW

**Comment by pupnewfster** [comment]
Adding a tag for shears (and adding the shears tag to the tools tag) might make sense for the purposes of modded shears.

**Comment by issork** [comment]
Added shears as well. Edit: Forgot to add the shears to the tool tag.

File: src/main/java/net/minecraftforge/common/data/ForgeItemTagsProvider.java [hunk 1]
```diff
@@ -141,6 +141,70 @@ public void registerTags()
...
+        getBuilder(Tags.Items.SWORDS_WOODEN).add(Items.WOODEN_SWORD);
+        getBuilder(Tags.Items.SWORDS_STONE).add(Items.STONE_SWORD);
+        getBuilder(Tags.Items.SWORDS_IRON).add(Items.IRON_SWORD);
+        getBuilder(Tags.Items.SWORDS_GOLDEN).add(Items.GOLDEN_SWORD);
+        getBuilder(Tags.Items.SWORDS_DIAMOND).add(Items.DIAMOND_SWORD);
+        getBuilder(Tags.Items.BOWS).add(Items.BOW);
+        getBuilder(Tags.Items.CROSSBOWS).add(Items.CROSSBOW);
+        getBuilder(Tags.Items.TRIDENTS).add(Items.TRIDENT);
+        
+        getBuilder(Tags.Items.TOOLS).add(Tags.Items.HOES, Tags.Items.AXES, Tags.Items.PICKAXES, Tags.Items.SHOVELS);
```
Comment (pupnewfster) [inline src/main/java/net/minecraftforge/common/data/ForgeItemTagsProvider.java]: This does not include the list of shears, even though they are in the sub directory/tag for tools.

Edit: I see you noticed that fact as I was typing this.

Comment (issork) [reply src/main/java/net/minecraftforge/common/data/ForgeItemTagsProvider.java]: Yes, I just noticed, on it ^^

**Comment by Draco18s** [comment]
Possibly outside the scope, but related:
Several loot tables (leaves, mostly) refer to `"item": "minecraft:shears"` which makes modded shears impractical. May be worth converting those over to `"tag": "forge:tools/shears"` instead.

**Comment by tterrag1098** [comment]
I'm not sure shields are "armor", they feel more like tools/weapons to me. A defensive weapon, but not armor.

**Comment by issork** [comment]
Shields are armor by definition, so I wouldn't feel very comfortable putting them in the tool/weapon category. I'd be fine with shields having their own top level tag though.

**Comment by Draco18s** [comment]
@tterrag1098 Shields provide damage reduction, same as worn armor. Close enough by any definition.

**Pull request closed without merging**